components/model_analyzer/TorchBenchAnalyzer.py

```diff
@@ -0,0 +1,120 @@
+    
+    def aggregate(self):
+        gpu_record_aggregator = RecordAggregator()
+        gpu_record_aggregator.insert_all(self.gpu_records)
+        # @Yueming Hao todo: groupby GPU, then metric. Then calculate the flops
+        records_groupby_gpu = {}
+        records_groupby_gpu = gpu_record_aggregator.groupby(
+            self.gpu_metrics, lambda record: record.device_uuid())
+        
+        for gpu in self.gpus:
+            self.gpu_metric_value[gpu.device_uuid()] = {}
+        for metric_type, metric in records_groupby_gpu.items():
+            for gpu_uuid, metric_value in metric.items():
+                self.gpu_metric_value[gpu_uuid][metric_type] = metric_value
+        # return self.gpu_metric_value
+    
+    def set_monitoring_interval(self, attempted_interval):
+        """
+        The default monitoring internval is 10ms.
+        """
+        if attempted_interval < 0.001:
+            logger.warning("The attempted interval is too short, would cause untruested profiling results.")
+        self.config.monitoring_interval = attempted_interval
+
+    def print_flops(self):
+        print("==========Summary==========")
+        for gpu_uuid in self.gpu_metric_value:
+            gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+            print(self.gpu_metric_value[gpu_uuid][GPUFP32Active].value())
+            # TFLOPs/second = Device_SM_Count x Device_FMAs_Per_Cycle_Per_SM x 2 x Running_Frequency_KHz x DCGM_Activity / 1e+9
+            print("GPU : TFLOPs/Second %.4f" % (gpu._sm_count * gpu._fma_count * 2 *
+                gpu._frequency * self.gpu_metric_value[gpu_uuid][GPUFP32Active].value() / 1e+9))
+        # @Yueming Hao: print all collected gpu records, for debug only
+        # print(self.gpu_metrics)
+        # for item in self.gpu_records:
+        #     print(item)
+        #     print(item.value())
+    
+    def calculate_flops(self, gpu_uuid=None):
+        """
+        The function to calculate TFLOPs/second for the desired GPU or the first available GPU.
+        @return : a floating number representing TFLOPs/second.
+        """
+        if gpu_uuid:
+            if gpu_uuid in self.gpu_metric_value:
+                gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+                return gpu._sm_count * gpu._fma_count * 2 * gpu._frequency * self.gpu_metric_value[gpu_uuid][GPUFP32Active].value() / 1e+9
+            else:
+                raise TorchBenchAnalyzerException("No available GPU with uuid ", gpu_uuid, " found!")
+        else:
+            if len(self.gpu_metric_value) > 1:
+                logger.warning("There are multiple available GPUs and will only return the first one's flops.")
+            gpu_uuid = next(iter(self.gpu_metric_value))
+            gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+            return gpu._sm_count * gpu._fma_count * 2 * gpu._frequency * self.gpu_metric_value[gpu_uuid][GPUFP32Active].value() / 1e+9
+
+def test_dcgm_installation():
```

Comment:
The prefix `test_` is for unit tests, which this is a runtime check, so better rename it to `check_dcgm()`.